src/libraries/Microsoft.PowerFx.Core/Public/Values/CollectionTableValue.cs

```diff
@@ -107,5 +107,51 @@ record = Marshal(item);
                 return base.TryGetIndex(index1, out record);
             }
         }
+
+        public override async Task<DValue<RecordValue>> PatchCoreAsync(RecordValue baserecord, RecordValue newRecord)
+        {
+            var actual = Find(baserecord);
+
+            if (actual != null)
+            {
+                return await actual.UpdateFields(newRecord);
+            }
+            else
+            {
+                return DValue<RecordValue>.Of(FormulaValue.NewBlank(IRContext.ResultType));
+            }
+        }
+
+        // Linear scan
```

Comment:
>// Linear scan [](http://example.com/codeflow?start=8&length=14)

we should comment these more, especially since it's virtual. 

Notably that it returns null on not found (as opposed to throws)